src/assertions/threshold.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    Threshold(ratio; weights=nothing)
+
+Assert that the ratio of missing values in the provided dataset does not exceed to specified ratio.
+If a weights array is provided then the ratio will be calculated as the
+`sum(weights[ismissing.(data)]) / sum(weights)`
+"""
+struct Threshold <: Assertion
+    ratio::Float64
+    weights::Union{AbstractVector, Nothing}
```

Comment:
Should this use `StatsBase.uweights` instead of `nothing`?
I am guessing not, because you want one that doens't need to know the size.
Which was my main complain in the PR that introducted unit weights: that it had a fixed length.

On the other hand, its also annoying for the case were we do know thge lengths to pass it in at the time we construct this object.

I wonder if we should look at this being allowed to take a function

Reply:
We might want to make this a type-parameter so the compiler can specialie on the `Nothing` case

Reply:
I suppose they could also just be separate type `Threshold` and `WeightedThreshold`, but then that'll require keeping the `Assertion` abstract type.

Reply:
Made an issue for that. https://github.com/invenia/Impute.jl/issues/74